fdbserver/TenantBalancer.actor.cpp

```diff
@@ -78,18 +87,27 @@ class SourceMovementRecord {
 	Standalone<StringRef> sourcePrefix;
 	Standalone<StringRef> destinationPrefix;
 
-	std::string databaseName;
+	std::string destDatabaseName;
+	// TODO: leave this open, or open it at request time?
 	Database destinationDb;
+	std::string tagName;
 };
 
 class DestinationMovementRecord {
 public:
 	DestinationMovementRecord() {}
-	DestinationMovementRecord(Standalone<StringRef> sourcePrefix, Standalone<StringRef> destinationPrefix)
-	  : id(deterministicRandom()->randomUniqueID()), sourcePrefix(sourcePrefix), destinationPrefix(destinationPrefix) {}
+	DestinationMovementRecord(Standalone<StringRef> sourcePrefix,
+	                          Standalone<StringRef> destinationPrefix,
+	                          std::string srcDatabaseName,
+	                          Database sourceDb)
+	  : id(deterministicRandom()->randomUniqueID()), sourcePrefix(sourcePrefix), destinationPrefix(destinationPrefix),
+	    srcDatabaseName(srcDatabaseName), sourceDb(sourceDb), tagName(DBMOVE_TAG_PREFIX + sourcePrefix.toString()) {}
 
 	Standalone<StringRef> getSourcePrefix() const { return sourcePrefix; }
 	Standalone<StringRef> getDestinationPrefix() const { return destinationPrefix; }
+	Database getSourceDatabase() const { return sourceDb; }
+	std::string getSrcDBName() const { return srcDatabaseName; }
```

Comment:
For consistency, let's use the full words source and destination.

```suggestion
	std::string getSourceDBName() const { return sourceDatabaseName; }
```